src/Illuminate/Database/Query/Builder.php

```diff
@@ -2703,11 +2723,63 @@ public function soleValue($column)
      */
     public function get($columns = ['*'])
     {
+        if (! is_null($keyBy = $this->keyBy)) {
+            return $this->getKeyed($keyBy, $columns);
+        }
+
         return collect($this->onceWithColumns(Arr::wrap($columns), function () {
             return $this->processor->processSelect($this, $this->runSelect());
         }));
     }
 
+    /**
+     * Execute the query as a "select" statement where the resulting array is keyed by the given column.
+     *
+     * @param  string  $key
+     * @param  array|string  $columns
+     * @return \Illuminate\Support\Collection
+     */
+    protected function getKeyed($key, $columns = ['*'])
+    {
+        return collect($this->onceWithColumns(Arr::wrap($columns), function () use ($key) {
+            $this->columns = Arr::prepend($this->qualifyStarColumns($this->columns), $key);
```

Comment:
does this work well when both the key is in the sql result set as first item and later on in the array? does that not lead to problems with column names or does PDO solve that?

Reply:
I couldnt think of possible issues. You can freely do `SELECT name, name, name FROM users` and you will get the expected result (3x the same column per record). The only thing that happens explicitly here is to add this column *within* the ` onceWithColumns` callback, and also to make sure that any star column is prefixed with a table name since `SELECT name, * FROM users` is not valid SQL.

Reply:
Just to be clear, `PDO::FETCH_UNIQUE` always uses the first column of the result set as the array key. If there are multiple rows/records with the same key, the first record will occupy the array and later values will be discarded.